sycl/doc/extensions/SRGB/srgb_support.asciidoc

```diff
@@ -0,0 +1,158 @@
+= SYCL_EXT_ONEAPI_SRGB
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+Copyright (c) 2021-2021 Intel Corporation.  All rights reserved.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension is written against the SYCL 1.2.1 specification, Revision v1.2.1-6.
+
+== Introduction
+
+sRGB isn't an image data format, per se. It is a linearization of RGB color values that adjusts the color to be better matched with a particular hardware medium. A more detailed explanation may be found here: https://software.intel.com/content/www/us/en/develop/articles/using-opencl-20-srgb-image-format.html
+
+OpenCL provides automatic linearization of RGB values into sRGB color space. OneAPI now exposes that OpenCL support for read_only images.
+
+== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_SRGB` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version.
+|===
+
+== aspect::ext_intel_oneapi_srgb
+Since not all devices and backends support sRGB images, `aspect::ext_intel_oneapi_srgb` is available to programmatically detect this.
+[source]
+----
+ // someDevice is a sycl::device
+ if(someDevice.has(aspect::ext_intel_oneapi_srgb)){ ... }
+----
+
+== New ImageChannelOrder Enums
+
+sRGB values have been added to the `image_channel_order` enumeration. The `image_channel_order::ext_oneapi_srgba` value allows images to be read with sRGB linearization of the color values. This value can be passed to `image` constructors, with some limitations that are outlined below.
+
+The values `ext_oneapi_srgb` and `ext_oneapi_srgbx` have also been added. `ext_oneapi_srgb` for 24-bit sRGB images, and `ext_oneapi_srgbx` for 24-bit images in 32-bit space (where the alpha channel is ignored). However, it should be noted that like their existing `rgb` and `rgbx` counterparts, neither is working at this time. They are added in hopes of future support
```

Comment:
If neither `rgb` nor `rgbx` are supported yet, then maybe it is better to omit them for this version of the extension.  We can always add a new revision of the extension later which adds more APIs.

Reply:
Well, `rgb` and `rgbx` are already in the enumeration, just not supported yet.  I have now added `ext_oneapi_srgb` and `ext_oneapi_srgbx` (the "S" variants of the same). If we fix/enable RGB and RGBX, we will very likely unblock SRGB and SRGBX as well.  The issue is subtle, "RGB" is in the spec, but there is some rule that says the data size has to be a power of 2, and 24 bit is not.  I made a note awhile ago to investigate this, but have never gotten to it.

Reply:
Sorry, I meant the `ext_oneapi_srgb` and `ext_oneapi_srgbx` enumerators.  It seems weird that we would add them to the enumeration even when they are not supported.  Why not wait until we have support, and then:

* Add them to the enumeration.
* Describe them in the extension spec.
* Bump the extension spec revision, so applications can test the revision number to know whether they are supported.

Reply:
ok. removed.